Add docs for storage version two (#70)
spheronFdn/docs: pages/sdk/storage-v2.mdx

```js
const { cdnARecords, cdnCnameRecords } = await client.getCdnDnsRecords();
```

Function `getCdnDnsRecords` doesn't take any parameters.

**Response**

- **cdnARecords** - (string) - will contain the DNS value that should be used for domains.
- **cdnCnameRecords** - (string) - will contain the DNS value that should be used for subdomains.

---

### Get Token Scope

Used to get the scope of the token.

```js
const tokenScope: TokenScope = await client.getTokenScope();
```

Function `getTokenScope` takes no parameters.

**Response**

```js
interface TokenScope {
  user: {
    id: string,
    username: string,
    name: string,
    email: string,
  };
  organizations: {
    id: string,
    name: string,
    username: string,
  }[];
}
```

---

## Storage Utility

### Transforming CID from V0 to V1 and vice versa

The package also provides a couple of methods for transforming CID from V0 to V1 and vice verse.

```js
import { ipfs } from "@spheron/storage";

const cid = <CID_VALUE>;

const v1 = ipfs.utils.toV1(cid);
console.log("CID V1", v1);

const v0 = ipfs.utils.toV0(cid);
console.log("CID V0", v0);

> <Navigation name="Storage SDK V2" />

Author: done

> <Navigation name="Storage SDK V2" />

Reviewer: Remove these -> ``` around Navigation.

> <Navigation name="Storage SDK V2" />

Author: fixed

import { Callout } from "nextra-theme-docs";
import Navigation from "components/navigation";

# Storage SDK

## Available Methods

```
CHANGED FUNCTIONS:

> getBucket, getBucketUploadCount, getBucketUploads, getOrganizationUsage, getUpload, all the ipns ones

Reviewer 2: mention the ipns methods as well rather than mention like this and mention it in bullet list format

Author: do0ne, check it out

New Methods:
- getOrganizationBuckets
- getOrganizationBucketCount
```

The `SpheronClient` instance provides several methods for working with buckets. Here's the list of all the supported methods.

### Organization

#### Migrate Buckets From Web Organization

> Used to migrate the Buckets and their Uploads from Web App organization to Storage organization.

Reviewer 2: not web app, make it Static Site

Author: done

From version `2.0.0` of `@spheron/storage`, the SDK will only work with the Storage organization. Using this method you can migrate the Buckets and Uploads you have used before.

> Executing the migration will move the following data from webapp to storage organization:

Reviewer 2: ```suggestion
Executing the migration will move the following data from static site to storage organization:
```

Author: done in all places

- Buckets
- Uploads
- Domains
- IPNS Records

<Callout type="info">
  **NOTE:** Important: After you migrate the data it is not possible to revert
  it.
</Callout>

```js

> const { numberOfBuckets, numberOfUploads } =

Reviewer 2: mention how to intialise the spheron client here as well. Also I'm confused how to safeguard the web app from where the data will be migrated since we are not taking it's token anywhere as per this doc, then anyone can migrate from any webapp if they know it's id

Author: There is a restriction on the BE. The user needs to be an owner on both the storage and web-app organization.

Author: Also added a note that the token for storage organization should be used, and that the user needs to be an owner of both organizations.

    ipfsNumberOfRequests: // the number of IPFS requests the organization has had till now
  };
  limit: {
    bandwidth: number, // the bandwidth limit for the current subscription
    storageArweave: number, // the limit in bytes for the Arweave storage for the current subscription
    storageIPFS: number, // the limit in bytes for the IPFS and Filecoin storage for the current subscription
    domains: number, // the limit on how many domains and subdomains can a organization have
    hnsDomains: number, // the limit on how many hns domains and hns subdomain can a organization have
    ensDomains: number, // the limit on how many ens domains a organization have
    parallelUploads: number, // the number on how many parallel uploads an organization can have
    imageOptimization: number, // the limit on how many optimized images can an organization have
    ipfsBandwidth: number; // the IPFS bandwidth limit for the current subscription
  };
}
```

> Changes, there are a couple of new properties, storageFilecoin, hns and ens domains ( not sure why they were missing) and optimized images

---

> ### Upload

Reviewer 2: put upload first, then org usage, ipns, bucket. update it in this order

Author: Done

```

Function `getBucket` takes just one parameter:

- **bucketId** - (string) - The id of the bucket.

**Response**

```js
enum  BucketStateEnum {
  MAINTAINED = "MAINTAINED",
  ARCHIVED = "ARCHIVED",
}

interface Bucket {
  id: string;
  name: string;
  organizationId: string;
  state: BucketStateEnum;
}
```

> Difference is no domains here, user can use the method to get domains

---

#### Archive Bucket

Used to archive the Bucket. New uploads cannot be created for an archived bucket.

```js
await client.archiveBucket(bucketId);
```

Function `archiveBucket` takes just one parameter:

- **bucketId** - (string) - The id of the bucket.

---

#### Unarchive Bucket

Used to unarchive the Bucket.

```js
await client.unarchiveBucket(bucketId);
```

Function `unarchiveBucket` takes just one parameter:

- **bucketId** - (string) - The id of the bucket.

---

#### Get Bucket Upload Count

Used to get the number of uploads for the specified bucket.

```js
const count: number = await client.getBucketUploadCount(bucketId);
```

Function `getBucketUploadCount` takes just one parameter:

- **bucketId** - (string) - The id of the bucket.

> ````

Reviewer 2: there is something wrong here with syntax, please check

Author: Fixed

**Response**

```js
enum DomainTypeEnum {
  DOMAIN = "domain",
  SUBDOMAIN = "subdomain",
  HANDSHAKE_DOMAIN = "handshake-domain",
  HANDSHAKE_SUBDOMAIN = "handshake-subdomain",
  ENS_DOMAIN = "ens-domain",
}

interface Domain {
  id: string;
  name: string;
  link: string;
  verified: boolean;
  bucketId: string;
  type: DomainTypeEnum;
}
```

---

#### Delete Bucket Domain

Used to delete the domain of the Bucket.

```js
await client.deleteBucketDomain(bucketId, domainIdentifier);
```

Function `deleteBucketDomain` takes two parameters:

- **bucketId** - (string) - The id of the bucket.
- **domainIdentifier** - (string) - It can either be the id of the domain or the name of the domain.

---

#### Get CDN DNS Records

Used to get the values that should be set on your DNS provider for your bucket domains.

```js
const { cdnARecords, cdnCnameRecords } = await client.getCdnDnsRecords();
```

Function `getCdnDnsRecords` doesn't take any parameters.

**Response**

- **cdnARecords** - (string) - will contain the DNS value that should be used for domains.
- **cdnCnameRecords** - (string) - will contain the DNS value that should be used for subdomains.

---

> ### Bucket Ipns Records

Reviewer 2: ```suggestion
### IPNS Records
```

Author: Done

> #### Get Bucket Ipns records

Reviewer 2: ```suggestion
#### Get Bucket IPNS records
```
do this everywhere

Author: Done

- Changes in methods and interfaces:
  - [getUpload](/sdk/storage-v2/#get-upload) - The `Upload` interface has changed. `buildDirectory` property is renamed to `uploadDirectory`, and now it is an array of with both the name and the size of the uploaded files. The `UploadStatusEnum` has also changed.
  - [getOrganizationUsage](/sdk/storage-v2/#get-organization-usage) - New properties have been added.
  - [getBucket](/sdk/storage-v2/#get-bucket) - The `Bucket` interface does not include the domains anymore. To fetch the domains of the bucket you can use the [getBucketDomains](/sdk/storage-v2/#get-bucket-domains) method.
  - [getBucketUploadCount](/sdk/storage/#get-bucket-upload-count) - The method now only returns the total number of uploads.
  - [getBucketUploads](/sdk/storage/#get-bucket-upload-count) - The `Upload` interface has changed. `buildDirectory` property is renamed to `uploadDirectory`, and now it is an array of with both the name and the size of the uploaded files. The `UploadStatusEnum` has also changed.
  - IPNS Records are now grouped on the bucket level. Checkout the [section](/sdk/storage-v2/#ipns-records) for more details.

> ## Available Methods

Reviewer: What is this empty heading for?

Author: removed,
